test/test_ops.py

```diff
@@ -689,6 +689,31 @@ def test_cumsum_zero_axis(self):
     helper_test_op([(0,3)], lambda x: torch.cumsum(x, dim=0), lambda x: Tensor.cumsum(x, axis=0))
     helper_test_op([(2,3,0)], lambda x: torch.cumsum(x, dim=2), lambda x: Tensor.cumsum(x, axis=2))
 
+  @unittest.skipIf(getenv("CLANG"), "broken on clang for some reason")
+  def test_weird_clang_bug(self):
+    helper_test_op([(512)], lambda x: torch.stack(torch.cummax(x, dim=0)), lambda x: Tensor.stack(*x.cummax(axis=0)))
+    helper_test_op([(1022)], lambda x: torch.stack(torch.cummax(x, dim=0)), lambda x: Tensor.stack(*x.cummax(axis=0)))
+    helper_test_op([(20,30,40)], lambda x: torch.stack(torch.cummax(x, dim=2)), lambda x: Tensor.stack(*x.cummax(axis=2)))
+    helper_test_op([(20,30,40)], lambda x: torch.stack(torch.cummax(x, dim=-1)), lambda x: Tensor.stack(*x.cummax(axis=-1)))
```

Comment:
these four tests are the four commented ones below. they all fail on `CLANG=1` with

```
----------------------------- Captured stderr call -----------------------------
<stdin>:4:18: error: integer literal is too large to be represented in a signed integer type, interpreting as unsigned [-Werror,-Wimplicitly-unsigned-literal]
    long acc0 = -9223372036854775808ll;
                 ^
1 error generated.
```

```
======================================================================
ERROR: test_cummax (test.test_ops.TestOps.test_cummax)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jeff/Documents/repos/tinygrad/test/test_ops.py", line 701, in test_cummax
    helper_test_op([(20,30,40)], lambda x: torch.stack(torch.cummax(x, dim=2)), lambda x: Tensor.stack(*x.cummax(axis=2)))
  File "/Users/jeff/Documents/repos/tinygrad/test/test_ops.py", line 31, in helper_test_op
    ret = tinygrad_fxn(*tst).realize()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/tensor.py", line 3581, in _wrapper
    ret = fn(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/tensor.py", line 228, in realize
    run_schedule(*self.schedule_with_vars(*lst), do_update_stats=do_update_stats)
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/engine/realize.py", line 223, in run_schedule
    for ei in lower_schedule(schedule):
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/engine/realize.py", line 216, in lower_schedule
    raise e
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/engine/realize.py", line 210, in lower_schedule
    try: yield lower_schedule_item(si)
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/engine/realize.py", line 194, in lower_schedule_item
    runner = get_runner(si.outputs[0].device, si.ast)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/engine/realize.py", line 163, in get_runner
    method_cache[ckey] = method_cache[bkey] = ret = CompiledRunner(replace(prg, dname=dname))
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/engine/realize.py", line 84, in __init__
    self.lib:bytes = precompiled if precompiled is not None else Device[p.dname].compiler.compile_cached(p.src)
                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/device.py", line 187, in compile_cached
    lib = self.compile(src)
          ^^^^^^^^^^^^^^^^^
  File "/Users/jeff/Documents/repos/tinygrad/tinygrad/runtime/ops_clang.py", line 15, in compile
    subprocess.check_output(['clang', '-shared', *self.args, '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib',
  File "/opt/homebrew/Cellar/python@3.12/3.12.7/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.7/Frameworks/Python.framework/Versions/3.12/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['clang', '-shared', '-march=native', '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib', '-', '-o', '/var/folders/wh/m0ckr3n10fjbfzq08k3j0hlr0000gn/T/tmp_yjonoru']' returned non-zero exit status 1.
```